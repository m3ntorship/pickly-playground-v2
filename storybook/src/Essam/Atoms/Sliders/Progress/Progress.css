/*************   */

/* Sliders */

/*************   */
.progress[value] {
  @apply bg-white border-none rounded-full w-13.7xxl overflow-hidden h-3 p-1;
}

.progress[value]::-webkit-progress-bar {
  @apply bg-white border-none rounded-xsm w-13xxxl;
}

.progress[value]::-webkit-progress-value {
  @apply bg-primary border-none rounded-full;
}

.progress[value="100"]::-webkit-progress-value {
  @apply bg-success border-none rounded-full;
}

.progress.vote-meter-1,
.progress.vote-meter-2,
.progress.vote-meter-3 {
  @apply transform -rotate-90;
}

.progress.vote-meter-1[value]::-webkit-progress-value {
  @apply bg-success;
}

.progress.vote-meter-2[value]::-webkit-progress-value {
  @apply bg-primary-shd5;
}

.progress.vote-meter-3[value]::-webkit-progress-value {
  @apply bg-error;
}

.progress.vote-meter-4,
.progress.vote-meter-5 {
  @apply w-4xl h-4xl;
}

.progress.vote-meter-4 .progress-value,
.progress.vote-meter-5 .progress-value {
  @apply stroke-7  fill-white origin-center transform rotate-90;
}

.progress.vote-meter-4 .progress-value {
  @apply stroke-primary;
  transition: all 0.5s ease-in-out;
}

.progress.vote-meter-5 .progress-value {
  @apply stroke-error;
  transition: all 0.5s ease-in-out;
}

/*************   */

/* Sliders */

/*************   */
