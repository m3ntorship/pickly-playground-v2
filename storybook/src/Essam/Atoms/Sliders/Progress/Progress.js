import React from "react";
import PropTypes from "prop-types";
import "./Progress.css";

export const Progress = ({ value, max, min, ...props }) => {
  return (
    <progress className="progress" value={value} min={min} max={max} {...props}>
      50/100
    </progress>
  );
};

Progress.propTypes = {
  value: PropTypes.number,
  min: PropTypes.number,
  max: PropTypes.number,
};

Progress.defaultProps = {
  value: 0,
  min: 0,
  max: 0,
};
