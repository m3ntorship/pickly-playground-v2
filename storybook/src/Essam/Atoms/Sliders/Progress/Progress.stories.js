import React from "react";
import { Progress } from "./Progress";

export default {
  title: "Essam/Atoms/Sliders/Progress",
  component: Progress,
  argTypes: {
    value: {
      control: {
        type: "range",
        min: 0,
        max: 100,
        step: 1,
      },
    },
  },
};

const Template = (args) => <Progress {...args} />;

export const progressBar = Template.bind({});
progressBar.args = {
  value: 10,
  min: 0,
  max: 100,
};
