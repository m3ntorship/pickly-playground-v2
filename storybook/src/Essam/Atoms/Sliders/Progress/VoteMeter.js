import React, { useEffect } from "react";
import className from "classnames";
import "./Progress.css";

export const VoteMeter = ({ value }) => {
  const meterClasses = className("progress", {
    "vote-meter-4": value >= 50,
    "vote-meter-5": value < 50,
  });
  const meterRef = React.createRef();
  useEffect(() => {
    console.log(meterRef);
    let radius = meterRef.current.r.baseVal.value;
    console.log("radius", radius);
    let circumference = radius * 2 * Math.PI;
    meterRef.current.style.strokeDasharray = circumference;
    const setProgress = (percent) => {
      meterRef.current.style.strokeDashoffset =
        circumference - (percent / 100) * circumference;
      console.log(meterRef.current.style.strokeDashoffset);
    };
    setProgress(value);
  });
  return (
    <svg className={meterClasses}>
      <circle
        ref={meterRef}
        r="31"
        cx="36"
        cy="36"
        className="progress-value"
      ></circle>
    </svg>
  );
};
