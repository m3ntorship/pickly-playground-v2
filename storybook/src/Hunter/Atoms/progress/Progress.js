import React from "react";
import PropTypes from "prop-types";

export const PROGRESS = {
  TYPE: {
    HORIZONTAL: "Horizontal",
    VERTICAL: "Vertical",
  },
  RANGE: {
    MIN: 0,
    MAX: 30,
    STEP: 1,
  },
};

export const Progress = ({ disabled, range, type }) => {
  const direction =
    type === PROGRESS.TYPE.HORIZONTAL
      ? `h-0.5 w-${range}`
      : type === PROGRESS.TYPE.VERTICAL
      ? `w-1 h-${range}`
      : "";
  const color =
    range < PROGRESS.RANGE.MAX / 3
      ? "bg-error "
      : range < PROGRESS.RANGE.MAX
      ? "bg-primary "
      : "bg-success";

  return (
    <>
      <div
        className={`rounded-full
          ${direction} 
          ${color}
          ${disabled && "opacity-25"}
          `}
      ></div>
    </>
  );
};
Progress.propTypes = {
  type: PropTypes.oneOf(["Horizontal", "Vertical"]),
  range: PropTypes.number,
  color: PropTypes.string,
};

export default Progress;
