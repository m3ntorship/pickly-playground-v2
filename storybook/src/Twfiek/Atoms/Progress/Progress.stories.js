import React from "react";
import Progress from "./Progress";

const defaultExport = {
  title: "Twfiek/Atoms/progress",
  component: Progress,
  argTypes: {
    progress: {
      control: {
        type: "range",
        min: 0,
        max: 100,
        step: 2,
      },
    },
  },
};

export default defaultExport;

const Template = (args) => <Progress {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  progress: 20,
  row: true,
  disabled: false,
};
