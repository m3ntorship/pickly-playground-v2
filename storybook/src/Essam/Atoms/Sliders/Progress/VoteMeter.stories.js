import React from "react";
import { VoteMeter } from "./VoteMeter";

export default {
  title: "Essam/Atoms/Sliders/VoteMeter",
  component: VoteMeter,
  argTypes: {
    value: {
      control: {
        type: "range",
        min: 0,
        max: 100,
        step: 1,
      },
    },
  },
};

const Template = (args) => <VoteMeter {...args} />;

export const VoteMeterBar = Template.bind({});
VoteMeterBar.args = {
  value: 90,
};
