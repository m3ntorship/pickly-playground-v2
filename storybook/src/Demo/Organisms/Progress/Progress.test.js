import React from "react";
import renderer from "react-test-renderer";
import Progress from "./Progress";

describe("Progress Component", () => {
  it("should render vertical red slider when progress is less than 50", () => {
    const tree = renderer.create(<Progress progress={49} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render vertical purple slider when progress is less than 100", () => {
    const tree = renderer.create(<Progress progress={99} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render vertical green slider when progress is 100", () => {
    const tree = renderer.create(<Progress progress={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render horizontal green slider when progress is 100", () => {
    const tree = renderer.create(<Progress progress={100} row />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render disabled horizontal and purple slider when progress is less than 100", () => {
    const tree = renderer
      .create(<Progress progress={90} row disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
