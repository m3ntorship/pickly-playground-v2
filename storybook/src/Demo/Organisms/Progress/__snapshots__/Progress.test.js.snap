// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Component should render disabled horizontal and purple slider when progress is less than 100 1`] = `
<div
  className="bg-primary h-1 rounded-full opacity-25"
  style={
    Object {
      "width": "90%",
    }
  }
/>
`;

exports[`Progress Component should render horizontal green slider when progress is 100 1`] = `
<div
  className="bg-success h-1 rounded-full "
  style={
    Object {
      "width": "100%",
    }
  }
/>
`;

exports[`Progress Component should render vertical green slider when progress is 100 1`] = `
<div
  className="bg-success w-2 rounded-full "
  style={
    Object {
      "height": "300px",
    }
  }
/>
`;

exports[`Progress Component should render vertical purple slider when progress is less than 100 1`] = `
<div
  className="bg-primary w-2 rounded-full "
  style={
    Object {
      "height": "297px",
    }
  }
/>
`;

exports[`Progress Component should render vertical red slider when progress is less than 50 1`] = `
<div
  className="bg-error w-2 rounded-full "
  style={
    Object {
      "height": "147px",
    }
  }
/>
`;
